tensorflow/core/kernels/mkl_quantize_op.cc

```diff
@@ -57,24 +244,100 @@ class MklQuantizeV2Op : public OpKernel {
+                                          mode_string + "'."));
+      round_mode_ = ROUND_HALF_TO_EVEN;
+    }
   }
 
-  ~MklQuantizeV2Op() {}
+  ~MklQuantizeV2Op() {
+    if (minfirst_input_ != nullptr) {
+      delete minfirst_input_;
+      minfirst_input_ = nullptr;
+    }
+  }
+
+  float* GetMinfirstInputBuf(int size) {
+    if (!minfirst_input_) {
+      minfirst_input_ = new float[size];
+      minfirst_input_size_ = size;
+    } else if (size != minfirst_input_size_) {
+      delete minfirst_input_;
+      minfirst_input_ = new float[size];
+      minfirst_input_size_ = size;
+    }
+
+    return minfirst_input_;
+  }
+
+  void Compute_Scalar(OpKernelContext* ctx, float min_range, float max_range) {
+    // TO-DO - Scalar support has to be added for SCALE mode
+    OP_REQUIRES(ctx, (mode_ == QUANTIZE_MODE_MIN_FIRST),
+                errors::InvalidArgument(
+                    "Scalar calculation in MKL is supported only for"
+                    "MIN_FIRST mode for now."));
+
+    auto cpu_engine = engine(engine::cpu, 0);
+    const Tensor& input = ctx->input(0);
+    const unsigned int src_idx = 0;
+    const Tensor& src_tensor = MklGetInput(ctx, src_idx);
+
+    MklDnnShape output_mkl_shape;
+    output_mkl_shape.SetMklTensor(false);
+
+    Tensor* output_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 0, &output_tensor, src_tensor.shape(),
+                              output_mkl_shape);
+    TensorShape min_tf_shape = {};
+    MklDnnShape min_mkl_shape;
+    min_mkl_shape.SetMklTensor(false);
+    Tensor* output_min_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 1, &output_min_tensor, min_tf_shape,
+                              min_mkl_shape);
+    TensorShape max_tf_shape = {};
+    MklDnnShape max_mkl_shape;
+    max_mkl_shape.SetMklTensor(false);
+    Tensor* output_max_tensor = nullptr;
+    AllocateOutputSetMklShape(ctx, 2, &output_max_tensor, max_tf_shape,
+                              max_mkl_shape);
+
+    // Estimate scale for qunatization
+    float scale_factor = 0;
+    const int number_of_bits = sizeof(T) * 8;
+    const int64 number_of_steps = static_cast<int64>(1) << number_of_bits;
+    scale_factor = (number_of_steps - 1.0) / (max_range - min_range);
+
+    float* src_data = const_cast<float*>(src_tensor.flat<float>().data());
+    T* out_data = output_tensor->flat<T>().data();
+
+    out_data[0] = (src_data[0] - min_range) * scale_factor;
+    output_min_tensor->flat<float>()(0) = min_range;
+    output_max_tensor->flat<float>()(0) = max_range;
+
+    return;
+  }
 
   void Compute(OpKernelContext* ctx) override {
+    const Tensor& input = ctx->input(0);
     const float input_min_range = ctx->input(1).flat<float>()(0);
     const float input_max_range = ctx->input(2).flat<float>()(0);
     float min_range = std::min(0.0f, input_min_range);
     float max_range;
-    OP_REQUIRES(ctx, (input_max_range > input_min_range),
+    OP_REQUIRES(ctx, !(input_max_range < input_min_range),
```

Comment:
Nit: I think the original expression is more straightforward and easier to understand.
```suggestion
    OP_REQUIRES(ctx, input_max_range > input_min_range,
```

Reply:
I changed this to fix a unit test where min and max ranges were same. In such scenarios, we need not send error message. Also I referred the code in quantize_op.cc which fixes this in the same way. Hope this make sense.

Reply:
Got it. Would this work? It's equivalent and more straightforward.
```suggestion
    OP_REQUIRES(ctx, input_max_range >= input_min_range,
```